edgelet/edgelet-kube/src/runtime.rs

```diff
@@ -432,9 +477,60 @@ where
     fn remove_all(&self) -> Self::RemoveAllFuture {
         Box::new(future::ok(()))
     }
+}
+
+impl<T, S> Authenticator for KubeModuleRuntime<T, S>
+where
+    T: TokenSource + Clone + 'static,
+    S: Service + Send + 'static,
+    S::ReqBody: From<Vec<u8>>,
+    S::ResBody: Stream,
+    Body: From<S::ResBody>,
+    <S::ResBody as Stream>::Item: AsRef<[u8]>,
+    <S::ResBody as Stream>::Error: Into<KubeClientError>,
+    S::Error: Into<KubeClientError>,
+    S::Future: Send,
+{
+    type Error = Error;
+    type Request = Request<Body>;
+    type AuthenticateFuture = Box<dyn Future<Item = AuthId, Error = Self::Error> + Send>;
+
+    fn authenticate(&self, req: &Self::Request) -> Self::AuthenticateFuture {
+        let token = req
+            .headers()
+            .get(header::AUTHORIZATION)
+            .and_then(|token| token.to_str().ok())
```

Comment:
The `.ok` call here is essentially saying that a failure to convert the header value into a string is the same as not specifying a token at all. Is that accurate? Should we fail instead?

Reply:
Good point.